src/main/java/com/cburch/logisim/std/wiring/PowerOnReset.java

```diff
@@ -103,50 +186,123 @@ public void actionPerformed(ActionEvent e) {
-    g.setFont(old.deriveFont(18.0f).deriveFont(Font.BOLD));
-    FontMetrics fm = g.getFontMetrics();
-    String txt = "Power-On Reset";
-    int wide = Math.max(bds.getWidth(), bds.getHeight());
-    int offset = (wide - fm.stringWidth(txt)) / 2;
-    Direction facing = painter.getAttributeValue(StdAttr.FACING);
-    if (((facing == Direction.NORTH) || (facing == Direction.SOUTH)) && (g instanceof Graphics2D g2)) {
-      int xpos = facing == Direction.NORTH ? x + 20 - fm.getDescent() : x + 20 + fm.getDescent();
-      int ypos = facing == Direction.NORTH ? y + offset : y + bds.getHeight() - offset;
-      g2.translate(xpos, ypos);
-      g2.rotate(facing.toRadians());
-      g.drawString(txt, 0, 0);
-      g2.rotate(-facing.toRadians());
-      g2.translate(-xpos, -ypos);
+    g.drawRect(x, y, width, height);
+
+    final var psize = painter.getAttributeValue(PORSIZE); 
+
+    if (psize == OLD_FORM) {
+      Font old = g.getFont();
+      g.setFont(old.deriveFont(16.0f).deriveFont(Font.BOLD));
+      String txt = S.get("porLongName"); 
+     
+      FontMetrics fm = g.getFontMetrics();  
+      int wide = Math.max(width, height);
+
+      int offset = (wide - fm.stringWidth(txt)) / 2;
+      Direction facing = painter.getAttributeValue(StdAttr.FACING);
+
+      if (((facing == Direction.NORTH) || (facing == Direction.SOUTH)) && (g instanceof Graphics2D g2)) {
+        int xpos = facing == Direction.NORTH ? x + 20 - fm.getDescent() : x + 20 + fm.getDescent();
+        int ypos = facing == Direction.NORTH ? y + offset : y + height - offset;
+        g2.translate(xpos, ypos);
+        g2.rotate(facing.toRadians());
+        g.drawString(txt, 0, 0);
+        g2.rotate(-facing.toRadians());
+        g2.translate(-xpos, -ypos);
+      } else {
+        g.drawString(txt, x + offset, y + fm.getDescent() + 20);
+      }
     } else {
-      g.drawString(txt, x + offset, y + fm.getDescent() + 20);
+      int x1;
+      int x2;
+      int x3;
+      int y1;
+      int y2;
+      int offset;
+      
+      Font old = g.getFont();
+      if  (psize == NARROW) {
+        g.setFont(old.deriveFont(6.0f).deriveFont(Font.BOLD));
+        offset = 7;
+      } else {
+        g.setFont(old.deriveFont(14.0f).deriveFont(Font.BOLD));
+        offset = 13;
+      }
+      
+      y1 = y + height - 4;
+      y2 = y + offset;
+      x1 = x + 3;
+      x2 = x + width - 4;
+      
+      Graphics2D g2 = (Graphics2D) g;
+      var oldStroke = g2.getStroke();
+      g2.setStroke(new BasicStroke(1));
+      g.setColor(Color.BLUE);
+      g.drawLine(x1, y1, x2, y1);
+      x1 = x1 + 1;
+      y1 = y1 + 1;
+      g.drawLine(x1, y2, x1, y1); 
+      g2.setStroke(oldStroke);
+      
+      x1 = x + 4;
+      x2 = x + width / 2;
+      x3 = x + width - 4;
+      y1 = y + offset + 2;
+      y2 = y + height - 5;
+
+      final var pstat = painter.getAttributeValue(PORTRANS); 
+      if (pstat == LTOH) {
+        y1 = y1 ^ y2 ^ (y2 = y1);  // swap y1 <-> y2
```

Comment:
Write this as:
var tmp = y1;
y1 = y2;
y2 = tmp;

It is better not to use unusual methods. Your expression leaves questions about the value of y2 when it is fetched. The optimizer will find the tmp easier to optimize.

Reply:
Ok, it's true that even for humans it's more understandable :-))